bindings/matrix-sdk-crypto-ffi/src/dehydrated_devices.rs

```diff
@@ -177,13 +180,13 @@ impl From<dehydrated_device::put_dehydrated_device::unstable::Request>
     }
 }
 
-fn get_pickle_key(pickle_key: &[u8]) -> Result<Box<[u8; 32]>, DehydrationError> {
+fn get_pickle_key(pickle_key: &[u8]) -> Result<DehydratedDeviceKey, DehydrationError> {
     let pickle_key_length = pickle_key.len();
 
     if pickle_key_length == 32 {
-        let mut key = Box::new([0u8; 32]);
-        key.copy_from_slice(pickle_key);
-
+        // safe to unwrap because we checked the size
+        let raw_bytes: [u8; 32] = pickle_key.try_into().unwrap();
```

Comment:
I would leave the copy from slice, `try_into()` moves `self` and moves are copies in Rust. Now here it's fine because you move a reference so we only copy the reference instead of the whole byte array.

But we always handle secret key material using `copy_from_slice()` and there it's obvious where the copy is made. The panic/unwrap is then also implicit and it's understood that you need a size check before.

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/4383/commits/c6a85e2322b8eb470f58cb3e485dc7955a2219a3